easybuild/easyblocks/t/tau.py

```diff
@@ -0,0 +1,278 @@
+            toolchain.CLANGGCC: ['clang', 'clang++', 'gfortran'],
+            toolchain.GCC: ['gcc', 'g++', 'gfortran'],
+            toolchain.INTELCOMP: ['icc', 'icpc', 'intel'],
+        }
+        comp_fam = self.toolchain.comp_family()
+        if comp_fam in known_compilers:
+            self.cc, self.cxx, self.fortran = known_compilers[comp_fam]
+        else:
+            raise EasyBuildError("Compiler family not supported yet: %s", comp_fam)
+
+        # determine values for MPI flags
+        self.mpi_inc_dir, self.mpi_lib_dir = os.getenv('MPI_INC_DIR'), os.getenv('MPI_LIB_DIR')
+        if self.mpi_inc_dir is None or self.mpi_lib_dir is None:
+            raise EasyBuildError("Failed to determine MPI include/library paths, no MPI available in toolchain?")
+
+        # determine value for optional packages option template
+        self.opt_pkgs_opts = ''
+        for dep, opt in [('PAPI', 'papi'), ('PDT', 'pdt'), ('binutils', 'bfd')]:
+            root = get_software_root(dep)
+            if root:
+                self.opt_pkgs_opts += ' -%s=%s' % (opt, root)
+
+        # determine list of labels, based on selected (extra) backends, variants and optional packages
+        self.variant_labels = []
+        backend_labels = ['', '-epilog-scalasca-trace', '-scorep', '-vampirtrace-trace']
+        for backend, backend_label in zip(['tau'] + KNOWN_BACKENDS.keys(), backend_labels):
+            if backend in ['tau'] + self.cfg['extra_backends']:
+                for pref, suff in [('-mpi', ''), ('', '-openmp-opari'), ('-mpi', '-openmp-opari')]:
+
+                    variant_label = 'tau'
+                    # For non-GCC builds, the compiler name is encoded in the variant
+                    if self.cxx != 'g++':
+                        variant_label += '-' + self.cxx
+                    if get_software_root('PAPI'):
+                        variant_label += '-papi'
+                    variant_label += pref
+                    if get_software_root('PDT'):
+                        variant_label += '-pdt'
+                    variant_label += suff + backend_label
+
+                    self.variant_labels.append(variant_label)
+
+        # make sure selected default TAU makefile will be available
+        avail_makefiles = ['Makefile.' + l for l in self.variant_labels]
+        if self.cfg['tau_makefile'] not in avail_makefiles:
+            raise EasyBuildError("Specified tau_makefile %s will not be available (only: %s)",
+                                 self.cfg['tau_makefile'], avail_makefiles)
+
+        # create install directory and make sure it does not get cleaned up again in the install step;
+        # the first configure iteration already puts things in place in the install directory,
+        # so that shouldn't get cleaned up afterwards...
+        self.log.info("Creating install dir %s before starting configure-build-install iterations", self.installdir)
+        mkdir(self.installdir, parents=True)
```

Comment:
TODO: remove install dir if it's already there (e.g. because of forced reinstallation)